neural_structured_learning/examples/notebooks/adversarial_cnn_transfer_learning_fashionmnist.ipynb

```diff
@@ -0,0 +1,1263 @@
+        "                            target_names=class_names))\n",
+        "pd.DataFrame(confusion_matrix(test_labels, prediction_labels), \n",
+        "             index=class_names, columns=class_names)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "mEqFDKAKy1xU"
+      },
+      "source": [
+        "#### Almost similar performance as our non-adversial trained CNN model!"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "oqIE9cTI-dbn"
+      },
+      "source": [
+        "# VGG-19 Base Model performance on Perturbed Test Dataset\n",
+        "\n",
+        "#### Let's look at the performance of our base VGG-19 model on the perturbed dataset."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "xXD4kcT2rtcI"
+      },
+      "source": [
+        "predictions = model.predict(perturbed_test_imgs)\n",
+        "prediction_labels = np.argmax(predictions, axis=1)\n",
+        "print(classification_report(test_labels, prediction_labels, \n",
+        "                            target_names=class_names))\n",
+        "pd.DataFrame(confusion_matrix(test_labels, prediction_labels), \n",
+        "             index=class_names, columns=class_names)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "U1p6jOOp-wMh"
+      },
+      "source": [
+        " We can see that the performance of the base VGG-19 (non adversarial-trained) model reduces by almost 50% on the perturbed test dataset, bringing a powerful ImageNet winning model to its knees!"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "VWAtWpi4-gfM"
+      },
+      "source": [
+        "# VGG-19 Adversarial Model performance on Perturbed Test Dataset\n",
+        "\n",
+        "### Evaluating our adverarial trained CNN model on the test dataset with perturbations. We see an approx. 38% jump in performance!"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "Xy3rG9lqU1Qr"
+      },
+      "source": [
+        "predictions = adv_model.base_model.predict(perturbed_test_imgs)\n",
+        "prediction_labels = np.argmax(predictions, axis=1)\n",
+        "print(classification_report(test_labels, prediction_labels, \n",
+        "                            target_names=class_names))\n",
+        "pd.DataFrame(confusion_matrix(test_labels, prediction_labels), \n",
+        "             index=class_names, columns=class_names)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "--jVksXQ-jKK"
+      },
+      "source": [
+        "# Compare Model Performances on Sample Perturbed Test Examples"
```

Comment:
Can you add a brief sentence or two here to summarize this cell?

Reply:
Sure done